src/app/autocorrect/autocorrect.component.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { AutocorrectComponent } from './autocorrect.component';
+import { AutoCompleteComponent } from '../auto-complete/auto-complete.component';
+import {RelatedSearchComponent} from "../related-search/related-search.component";
+import {InfoboxComponent} from "../infobox/infobox.component";
+import {ModalComponent} from "ng2-bs3-modal/components/modal";
+import {ContactComponent} from "../contact/contact.component";
+import {AboutComponent} from "../about/about.component";
+import {FooterNavbarComponent} from "../footer-navbar/footer-navbar.component";
+import {AppComponent} from "../app.component";
+import {NavbarComponent} from "../navbar/navbar.component";
+import {IndexComponent} from "../index/index.component";
+import {ResultsComponent} from "../results/results.component";
+import {NotFoundComponent} from "../not-found/not-found.component";
+import {AdvancedsearchComponent} from "../advancedsearch/advancedsearch.component";
+import {SearchBarComponent} from "../search-bar/search-bar.component";
+import {StoreDevtoolsModule} from "@ngrx/store-devtools";
+import {reducer} from "../reducers/index";
+import {StoreModule} from "@ngrx/store";
+import {JsonpModule, HttpModule} from "@angular/http";
+import {FormsModule} from "@angular/forms";
+import {CommonModule} from "@angular/common";
+import {BrowserModule} from "@angular/platform-browser";
+import {RouterTestingModule} from "@angular/router/testing";
+import { ThemeComponent } from '../theme/theme.component';
+import { DropdownComponent } from '../dropdown/dropdown.component';
+import {IntelligenceComponent} from "../intelligence/intelligence.component";
```

Comment:
import only, `AutocorrectService`, `RouterTestingModule`. It may throw errors like no provider for http/jsonp/store. In that case, import `HttpModule`, `JsonpModule`, `reducer`, `StoreModule`